src/commands/dev/socket/client.rs

```diff
@@ -0,0 +1,84 @@
+use chrome_devtools::events::DevtoolsEvent;
+use console::style;
+
+use openssl::ssl::{SslConnector, SslMethod, SslStream, SslVerifyMode};
+
+use ws::util::TcpStream;
+use ws::{Handler, Handshake, Message as WsMessage, Sender};
+
+use url::Url;
+
+pub struct WsClient {
+    pub out: Sender,
+}
+
+impl Handler for WsClient {
+    fn on_open(&mut self, _: Handshake) -> ws::Result<()> {
+        let enable_profiler = WsMessage::text(
+            r#"{
+                "id": 1,
+                "method": "Profiler.enable"
+            }"#,
+        );
+        let enable_runtime = WsMessage::text(
+            r#"{
+                "id": 2,
+                "method": "Runtime.enable"
+            }"#,
+        );
+        let enable_debugger = WsMessage::text(
+            r#"{
+                "id": 3,
+                "method": "Debugger.enable"
+            }"#,
+        );
+        self.out.send(enable_profiler)?;
+        self.out.send(enable_runtime)?;
+        self.out.send(enable_debugger)
+    }
+
+    fn on_message(&mut self, msg: WsMessage) -> ws::Result<()> {
+        let msg = msg.as_text()?;
+        log::info!("{}", msg);
+        let msg: Result<DevtoolsEvent, serde_json::Error> = serde_json::from_str(msg);
+        match msg {
+            Ok(msg) => {
+                match msg {
+                    DevtoolsEvent::ConsoleAPICalled(event) => match event.log_type.as_str() {
+                        "log" => println!("{}", style(event).blue()),
+                        "error" => eprintln!("{}", style(event).red()),
+                        _ => println!("unknown console event: {}", event),
```

Comment:
is this unreachable? are there other console event types? should the above strings be made into a deserializable enum?

Reply:
this is unfinished devtools response stuff

Reply:
and yes - i'm planning on making the log_types an enum in the devtools crate

Reply:
do we have a ticket to track the completion of this devtools response work? We should if we want to merge this PR into your feature branch.